---
title: "<i>P. vivax</i> single cell RNA seq: comparison of count data with and without UTR coordinates"
subtitle: "File: 1b_comparisonUTR.Rmd"
author: "Anthony Ruberto and Caitlin Bourke"
date: "`r Sys.Date()`"
output: html_document
---
```{r setup}
library(tidyverse)
library(Seurat)
library(scater)
library(scran)
library(SingleCellExperiment)
library(DropletUtils)
library(rtracklayer)
library(svglite)
library(gridExtra)
library(gridGraphics)
library(grid)
library(ggplotify)
library(patchwork)
library(GGally)
library(ggspatial)
library(rnaturalearth)
library(rnaturalearthdata)
library(rgeos)
library(rnaturalearthhires)
library(readxl)
library(data.table)
library(viridis)
library(ggsci)
source("scripts/functions_aesthetics.R")
source("scripts/geneInfo.R")
```


```{r import filtered from script1}
withUTR <- readRDS("outputs/1a_all_Pv_filtered.rds")
withUTR[[1]]@colData
```

```{r import raw noUTR files}

# save the Pv*_noUTR raw file output from STARsolo into folder "data" or provide path in replace of "data" for where to find these files.

noUTR_pv1 <-"data/Pv1_noUTR/raw/"
noUTR_pv2 <- "data/Pv3_noUTR/raw/"
noUTR_pv3 <- "data/Pv4_noUTR/raw/"

#these are the un-processed files direct from STARsolo - they contain all barcodes and are not subset. We will subset these to match the same cell barcodes as those that have been retained in the filtered object ("outputs/1_all_Pv_filtered.rds") for direct comparison 

Pv1noUTR<-read10xCounts(noUTR_pv1, col.names = T)
Pv2noUTR<-read10xCounts(noUTR_pv2, col.names = T)
Pv3noUTR<-read10xCounts(noUTR_pv3, col.names = T)

noUTRs <- list(Pv1noUTR, Pv2noUTR, Pv3noUTR)
```

```{r prepare colData of no UTR so should be the same}
case_ids <- c("Case_909", "Case_922", "Case_923")

for (i in 1:length(noUTRs)) {
 colData(noUTRs[[i]])$Sample <- paste0(case_ids[[i]])
 colData(noUTRs[[i]])$Barcode<-paste(colData(noUTRs[[i]])$Barcode,"_",colData(noUTRs[[i]])$Sample, sep = "")
}
```

```{r vector of retained barcodes from UTR analysis}
# the noUTR object still as rRNA genes which have already been removed from the tidied objects sceObjectFilter
rRNA_pv <- import.gff3("genefiles/PlasmoDB-51_PvivaxP01.gff") %>% 
  as_tibble() %>% 
  filter(type=="rRNA") %>% 
  select(ID, description) %>% 
  mutate(ID = gsub("\\.1", "", ID))

for (i in 1:length(noUTRs)) {
   noUTRs[[i]] <- noUTRs[[i]][!rownames(noUTRs[[i]])%in%rRNA_pv$ID,]
}

#nrows is now 6811

for (i in 1:length(noUTRs)) {
  noUTRs[[i]] <- noUTRs[[i]][,(colData(noUTRs[[i]])$Barcode)%in%(colData(withUTR[[i]])$Barcode)]
}
#creating a subset object of cells matched from the already processed 'withUTR' sceObjectFilter file to the unsubset noUTR files
```


```{r check sce objects without utr}
noUTRs[[1]]@colData
```
```{r remove genes from no UTR for equal comp with withUTR}
for (i in 1:length(noUTRs)){
  noUTRs[[i]] <- noUTRs[[i]][rowSums(counts(noUTRs[[i]])>2)>1, ]
}
#already done withUTRs...
```

```{r}
rowData(withUTR[[1]])
rowData(noUTRs[[1]])
```
```{r make long list of vectors of geneids from the two sce object lists}
geneids_noutr <- NULL
for (i in 1:length(noUTRs)) {
  geneids_noutr[[i]] <- rownames(noUTRs[[i]])
}

geneids_utr <- NULL
for (i in 1:length(withUTR)) {
  geneids_utr[[i]] <- rownames(withUTR[[i]])
}

```

```{r unique geneids in combined with utrs}
length(unique(c(geneids_utr[[1]], geneids_utr[[2]], geneids_utr[[3]])))

```
There are `r length(unique(c(geneids_utr[[1]], geneids_utr[[2]], geneids_utr[[3]])))` unique genes detected across the three replicates with the UTR info

```{r unique gene ids combined without utrs}
length(unique(c(geneids_noutr[[1]], geneids_noutr[[2]], geneids_noutr[[3]])))
```
There are `r length(unique(c(geneids_noutr[[1]], geneids_noutr[[2]], geneids_noutr[[3]])))` unique genes detected across the three replicates WITHOUT the UTR info.


```{r}
#This is a list of the genes that are detected in the datasets with UTRs
geneids_utr_vector <- unique(c(geneids_utr[[1]], geneids_utr[[2]], geneids_utr[[3]]))

#List from no UTRs
geneids_noUTR_vector <- unique(c(geneids_noutr[[1]], geneids_noutr[[2]], geneids_noutr[[3]]))

UTR_only <- geneids_utr_vector[!geneids_utr_vector%in%geneids_noUTR_vector] 

#There are also some genes that are detected only in the noUTR vector list - there are 11  - this is likely due to overlapping genes
geneids_noUTR_vector[!geneids_noUTR_vector%in%geneids_utr_vector] 
```


```{r make data frames of the pseudobulk counts}
pseudo_noutr <- NULL

for (i in 1:length(noUTRs)) {
  pseudo_noutr[[i]] <-counts(noUTRs[[i]]) %>% 
  as_tibble(rownames = "gene") %>% 
  pivot_longer(-gene) %>% 
  group_by(gene) %>% 
  summarise(sum_noUTR = sum(value), mean_noUTR = mean(value), median_noUTR = median(value)) %>% 
  mutate(rank_noUTR = dense_rank(-sum_noUTR)) %>% 
  arrange(rank_noUTR)
}

pseudo_withutr <- NULL
for (i in 1:length(withUTR)) {
  pseudo_withutr[[i]] <- counts(withUTR[[i]]) %>% 
  as_tibble(rownames = "gene") %>% 
  pivot_longer(-gene) %>% 
  group_by(gene) %>% 
  summarise(sum_withUTR = sum(value), mean_withUTR = mean(value), median_withUTR = median(value)) %>% 
  mutate(rank_withUTR = dense_rank(-sum_withUTR)) %>% 
  arrange(rank_withUTR)
}

#median is a bit useless with such a dropout rate...keeping for now
#is denserank so values with the same value have the same rank
```

```{r merge pseudobulk counts}
pv1 <- pseudo_withutr[[1]] %>% 
  full_join(pseudo_noutr[[1]]) %>% 
  mutate(rep = "Pv1")

pv2 <- pseudo_withutr[[2]] %>% 
  full_join(pseudo_noutr[[2]]) %>% 
  dplyr::mutate(rep = "Pv2")

pv3 <- pseudo_withutr[[3]] %>% 
  full_join(pseudo_noutr[[3]]) %>% 
   dplyr::mutate(rep = "Pv3")

all <- pv1 %>% 
  bind_rows(pv2) %>% 
  bind_rows(pv3) %>% 
  pivot_longer(cols = 2:9, names_to = "cond", values_to = "value") %>% 
  mutate(summ_stat = case_when(
    str_detect(cond, "sum") ~ "sum",
    str_detect(cond, "mean") ~ "mean",
    str_detect(cond, "median") ~ "median",
    str_detect(cond, "rank") ~ "rank"
  )) %>% 
  mutate(type = case_when(
    str_detect(cond, "with") ~ "withUTR",
    TRUE~"noUTR"
  ))

all[is.na(all)] <- 0
```

```{r create summary values across datasets}
all %>% 
  dplyr::filter(summ_stat=="sum") %>% 
  dplyr::filter(type=="withUTR") %>% 
  select(-c(cond, type)) %>% 
  pivot_wider(names_from = "summ_stat") %>%
  group_by(gene) %>% 
  summarise(sum_mean = mean(sum)) %>% 
  mutate(rank_mean = dense_rank(-sum_mean))


all_v2 <- all %>% 
  filter(type=="withUTR") %>% 
  filter(summ_stat=="sum") %>% 
  group_by(gene) %>% 
  mutate(mean_over_reps = mean(value)) %>% 
  mutate(total_sum = sum(value)) %>%
  select(-c(summ_stat, type)) %>% 
  pivot_wider(names_from = "rep")



all_v2$ranking_mean <- dense_rank(-all_v2$mean_over_reps)
all_v2$ranking_pv1 <- dense_rank(-all_v2$Pv1)
all_v2$ranking_pv2 <- dense_rank(-all_v2$Pv2)
all_v2$ranking_pv3 <- dense_rank(-all_v2$Pv3)
all_v2$total_sum_rank <- dense_rank(-all_v2$total_sum)

all_v2 %>% 
  write.csv("outputs/Pv10x_pseudobulk_ranking.csv")

```

```{r write supplementary file S3}
dir.create("tables")
all %>% 
  mutate(mod = paste0(rep, cond)) %>% 
  select(gene, value, mod) %>% 
  pivot_wider(names_from = mod) %>% 
  write.csv("tables/S3.csv")
```


```{r write supplementary file S4}
all %>% 
  filter(cond=="sum_withUTR") %>% 
  select(gene, rep, value) %>% 
  pivot_wider(names_from = "rep") %>% 
  mutate(detected = case_when(
    Pv1>0&Pv2>0&Pv3>0~"all",
    Pv1>0&Pv2>0&(is.na(Pv3)|Pv3<=0)~"Pv1&Pv2",
    Pv1>0&(is.na(Pv2)|Pv2<=0)&Pv3>0~"Pv1&Pv3",
    (is.na(Pv1)|Pv1<=0)&Pv2>0&Pv3>0~"Pv2&Pv3",
    Pv1>0&(is.na(Pv2)|Pv2<=0)&(is.na(Pv3)|Pv3<=0)~"Pv1",
    (is.na(Pv1)|Pv1<=0)&Pv2>0&(is.na(Pv3)|Pv3<=0)~"Pv2",
    (is.na(Pv1)|Pv1<=0)&(is.na(Pv2)|Pv2<=0)&Pv3>0~"Pv3"
  )) %>% 
  arrange(detected) %>% 
  drop_na(detected) %>% 
  write_csv("tables/S4.csv")
```


*** 
## Generating figures for panel  Figure 1

```{r Generating figures for panel 1}
fig_genes_UTR_detected <- all %>% 
  filter(summ_stat =="sum") %>% 
  mutate(summary = case_when(
    value>0 ~ 1,
    TRUE~ 0
  )) %>% 
  group_by(rep, cond) %>% 
  summarise(sum(summary)) %>% 
  ggplot(aes(x = rep, y =`sum(summary)`, fill = cond))+
    geom_col(position = "dodge")+
  labs(title = "Number of genes confidently identified", y = "total genes detected (post filter)", x = "")+
  scale_fill_manual(values = c("#808080","#000000"))+
  theme_linedraw()+
  tenx_theme()
```


```{r figure 1 increase in gene countrs, fig.height=3, fig.width=6}
genes_to_plot <- c("PVP01_1258000", "PVP01_0117400", "PVP01_0715100", "PVP01_1011800")

fig_UTR_increase <- all %>% 
  filter(summ_stat=="sum") %>% 
  filter(gene%in%genes_to_plot) %>% 
  mutate(gene = case_when(
    gene=="PVP01_1258000" ~ "PVP01_1258000::GEST",
    gene=="PVP01_0117400" ~ "PVP01_0117400::CNA",
    gene=="PVP01_0715100"~ "PVP01_0715100::HoMu",
    gene=="PVP01_1011800"~ "PVP01_1011800::unknown"
  )) %>% 
  mutate(gene = factor(gene, levels = c(
    "PVP01_1258000::GEST",
    "PVP01_0117400::CNA",
   "PVP01_0715100::HoMu",
   "PVP01_1011800::unknown"
  ))) %>% 
  group_by(rep) %>% 
  ggplot(aes(x = cond, y = value))+
  geom_boxplot(fill = "black", alpha = 0.2)+
  geom_point()+
  geom_line(aes(group = rep), alpha = 0.5)+
  theme_linedraw()+
  facet_wrap(~gene,  scales = "free", ncol = 5)+
  scale_x_discrete(labels = c("-", "+"))+
  labs(y = "Pseudobulk expression", x = "Transcriptome")+
  tenx_theme()+
  theme(axis.text.x = element_text(size = 30),
        panel.grid.major.y  = element_line(colour = "lightgrey"))
  

```

```{r figures for panel 1 import meta info}
meta <- NULL
for (i in 1:length(withUTR)) {
  meta[[i]] <- withUTR[[i]]@colData %>% as_tibble
}

meta <- meta[[1]] %>% 
  bind_rows(meta[[2]]) %>% 
  bind_rows(meta[[3]]) 
```

```{r number of cells}
cell_count <- meta %>% 
  ggplot(aes(x = Replicate))+
  geom_bar(stat = "count", fill = "black")+
    theme(legend.position = "none",
          axis.title.x = element_blank(),
          axis.text.x = element_text(angle = 90),
          axis.ticks.x  = element_blank())+
  labs(y = str_wrap("Number of sporozoites",15), x = "")+
  scale_x_discrete(labels = c(Pv1 = "Rep 1", Pv2 = "Rep 2", Pv3 = "Rep 3"))+
  tenx_theme()+
  theme(panel.grid.major.y = element_line(colour = "grey"))
```


```{r figure 1 proportion of reads mapping, fig.height=2, fig.width=4}
#mapping for replicates 

Rep1 <- c(0.474746, 0.250756, 0.191396)
Rep2 <- c(0.441397, 0.222121, 0.154148)
Rep3 <- c(0.518784, 0.302947, 0.24209)

mapping <- rbind(Rep1,
Rep2,
Rep3) 

colnames(mapping) <- c("Genome", "Transcriptome with UTR", "Transcriptome without UTR")

mapping_plot <- mapping %>% 
  as_tibble(rownames = "Replicate") %>% 
  pivot_longer(-Replicate) %>% 
  mutate(value = value * 100) %>% 
  ggplot(aes(x = Replicate, y = value, fill = name))+
  geom_col(position = "dodge")+
  coord_flip()+
  theme_linedraw()+
  scale_fill_manual(values = c("black", "#545454", "#a9a9a9"))+
  scale_x_discrete(limits = rev)+
  labs(x = "", y = "Percentage reads mapping", fill = "")+
  guides(fill=guide_legend(nrow=3,byrow=TRUE))+
  tenx_theme()+
  theme(panel.grid.major.x = element_line(colour = "grey"),
        legend.position = "bottom")
```

```{r plot intersection of genes mapping}
intersection_three_rep_geneids <- all %>% 
  filter(type=="withUTR") %>% 
  filter(summ_stat=="sum") %>% 
  filter(value>0) %>% 
  group_by(gene) %>% 
  dplyr::count() %>% 
  filter(n==3) 

mean_rank_to_plot <- all %>% 
  filter(summ_stat=="sum") %>% 
  filter(type=="withUTR") %>% 
  select(-c(cond, type)) %>% 
  pivot_wider(names_from = "summ_stat") %>%
  group_by(gene) %>% 
  summarise(sum_mean = mean(sum)) %>% 
  mutate(rank_mean = dense_rank(-sum_mean)) %>% 
  mutate(intersect = case_when(
    gene%in%intersection_three_rep_geneids$gene ~ "intersect",
    TRUE~ "not"
  ))


#this is a dense ranking so the max value is less than the number of values (if there are duplicates)
ranking <- ggplot()+
  geom_point(data = filter(mean_rank_to_plot, intersect=="not"), aes(x = rank_mean, y = sum_mean), alpha = 0.2)+
  geom_point(data = filter(mean_rank_to_plot, intersect=="intersect"), aes(x = rank_mean, y = sum_mean), alpha = 0.2, colour = "red")+
  scale_y_log10()+
  labs(x = "Gene ranking",
       y = str_wrap("Mean pseudobulk expression",15))+
  tenx_theme()+
  theme(panel.grid.major = element_line(colour = "grey"))
```


```{r create venn diagram of gene detected between reps}
rownames(withUTR[[1]])
genes_per_dataset <- NULL
  for (i in 1:length(withUTR)) {
  genes_per_dataset[[i]] <- rownames(withUTR[[i]])
}
venn::venn(
  genes_per_dataset,
  opacity = .3,
  zcolor = "grey50",
  cexsn = 0,
  cexil = 1,
  lwd = 2,
  col = "black",
  frame = T,
  borders = F,
  ggplot = F,
  ellipse = F,
  ilcs = 1,
  box = F,
  snames = c("Rep 1", "Rep 2", "Rep 3"),
  ilabels = F,
  margin(0)
)

grid.echo()
venn_plot <- grid.grab()
venn_plot <- as.ggplot(venn_plot)
dir.create("plots")
ggsave("plots/fig1e_venn.png", width = 5, height = 3)
```
```{r create corr plot between replicates fig1f}
sum_wide <- all %>% 
  filter(type=="withUTR") %>% 
  filter(summ_stat=="sum") %>% 
  select(gene, rep, value) %>% 
  pivot_wider(names_from = "rep") 

my_fn <- function(data, mapping, ...){
  p <- ggplot(data = data, mapping = mapping) + 
    geom_point(size = 0.5, alpha = 0.3, color = "black")+ 
    geom_smooth(method=lm, fill="red", color="red")+
    scale_x_log10()+
    scale_y_log10()
  p
}
correltion_fig1f <- ggpairs(sum_wide[,-1], 
        lower= list(continuous = my_fn), 
        upper = list(continuous = wrap(ggally_cor, displayGrid = T)), 
        proportions = "auto")+
  scale_x_log10()+
  scale_y_log10()+
  tenx_theme()

corr_grid <- ggpairs(sum_wide[,-1], 
        lower= list(continuous = my_fn, tenx_theme()),
        upper = list(continuous = wrap(ggally_cor, displayGrid = F),
                     theme(panel.grid.major = element_blank())), 
        proportions = "auto")+
  scale_x_log10()+
  scale_y_log10()+
  tenx_theme()

plots = list()
for (i in 1:9){
  plots <- c(plots, lapply(1:corr_grid$ncol, function(j) getPlot(corr_grid, i = i, j = j)))
}  

plots[[1]] <- plots[[1]]+tenx_theme()
plots[[2]] <- plots[[2]]+tenx_theme()
plots[[3]] <- plots[[3]]+tenx_theme()

plots[[4]] <- plots[[4]]+tenx_theme()+theme(panel.grid.major = element_line(colour = "grey"))
plots[[7]] <- plots[[7]]+tenx_theme()+theme(panel.grid.major = element_line(colour = "grey"))
plots[[8]] <- plots[[8]]+tenx_theme()+theme(panel.grid.major = element_line(colour = "grey"))

plots[[5]] <- plots[[5]]+tenx_theme()
plots[[6]] <- plots[[6]]+tenx_theme()
plots[[9]] <- plots[[9]]+tenx_theme()

correlation_fig1e <- ggmatrix(plots, 
         nrow = 3, 
         ncol=3, 
         xAxisLabels = c("Rep 1", "Rep 2", "Rep 3"), 
         yAxisLabels = c("Rep 1", "Rep 2", "Rep 3"))+
  theme(strip.text = element_text(family = "sans", size=15, colour = "white", margin = margin(0.2,0,0.2,0, "cm")),
      strip.text.y  = element_text(family = "sans", size=15, colour = "white", margin = margin(0,0.2,0,0.2, "cm"), angle = -90),
      strip.background = element_rect(colour = "black", fill = "black"))

ggsave("plots/fig1f.png", correlation_fig1e, width = 8, height = 7)
```
***
```{r create ggplot object map of cambodia}
asia <- ne_countries(continent = "Asia", returnclass = "sf", scale = "large")

#Kaev Seima coord = 106.8187, 12.3628 and Phnom Penh coord = 104.921111, 11.569444

map <- ggplot(data = asia)+
  geom_sf()+
   xlab("Longitude") + ylab("Latitude") +
  annotation_scale(location = "bl", width_hint = 0.3) +
  coord_sf(xlim = c(101, 109), ylim = c(10, 15.5))+
  annotate(geom = "text", x = 106.8, y = 12.6, label = "Kaev Seima", 
  fontface = "italic", color = "grey22", size = 4) +
  annotate(geom = "point", x = 106.8187, y = 12.3628, color = "grey22", size = 1) +
  annotate(geom = "point", x = 104.921111, y = 11.569444, color = "grey22", size = 1) +  
  annotate(geom = "text", x = 104.921111, y = 11.8, label = "Phnom Penh", 
  fontface = "italic", color = "grey22", size = 4) +
  annotate(geom = "text", x = 105, y = 13.2, label = "CAMBODIA", 
  fontface =  "bold", color = "grey22", size = 4) +
    annotate(geom = "text", x = 107.6, y = 11.5, label = "VIETNAM", 
  fontface =  "bold", color = "grey22", size = 4) +
      annotate(geom = "text", x = 106.5, y = 15, label = "LAOS", 
  fontface =  "bold", color = "grey22", size = 4) +
  annotate(geom = "text", x = 103, y = 15, label = "THAILAND", 
  fontface =  "bold", color = "grey22", size = 4) +
  theme_linedraw()+
  theme(panel.grid = element_blank(),
        axis.text = element_blank(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        plot.margin = unit(c(0.1,0.1,0.1,0.1), "cm")
        )
```

*** heatmap geom_tile

```{r Jex Westenberger Pv spz analysis}

Jex_RNA <- read_excel("genefiles/supplementary/Jex2019IJP_SupplementaryD3.xlsx", sheet = "S2")
Jex_RNA <- as.data.table(Jex_RNA) %>% 
  mutate(rank = dense_rank(-`mean TPMs`))

Jex_RNA [ , decile := cut(rank, breaks = quantile(rank, probs = seq(0, 1, by = 1/10)), labels = 1:10, right = FALSE)]
Jex_RNA<-Jex_RNA %>% 
  replace(is.na(.), 10) %>% 
  select(`Pv_P01 ID`, rank, decile) %>% 
  dplyr::rename(gene=`Pv_P01 ID`, spz_bulk_jex = rank) %>% 
  pivot_longer(cols = 2, names_to = "rep", values_to = "value")

#Read microarray data from AJ's reprocessed data as this is using P01 reference
Westenberger <- read_excel("genefiles/supplementary/Jex2019IJP_SupplementaryD3.xlsx", sheet = "S3")

Westenberger <-as.data.table(Westenberger) %>% 
  mutate(rank = dense_rank(-PvSPZ.microarray.E.mean.rank)) %>% 
  select(1,7,8) %>% 
  mutate( decile = cut(PvSPZ.microarray.E.mean.rank, breaks = quantile(PvSPZ.microarray.E.mean.rank, probs = seq(0, 1, by = 1/10)), labels = 1:10, right = FALSE)) %>% 
  replace(is.na(.), 10) %>% 
  dplyr::rename(gene=Pv_P01.ID) %>% 
  select(-PvSPZ.microarray.E.mean) %>% 
  pivot_longer(cols = 2, names_to = "rep", values_to = "value")

```

We can now look filter each dataset to assess the percentile for each of the UIS genes in PV.
Let's first provide a list of known UIS markers. These were obtained by taking the orthologs from P.berghei.


```{r integrate ranks}
tile_plot <- all %>% 
  filter(cond=="rank_withUTR") %>% 
  select(gene, rep, value) %>% 
  group_by(rep) %>% 
  mutate(decile := cut(value, breaks = quantile(value, probs = seq(0, 1, by = 1/10)), labels = 1:10, right = FALSE) ) %>% 
  bind_rows(Westenberger, Jex_RNA ) %>% 
  ungroup()

# 
# write.table(Pv_percentile, file = "outputs/PvUIS_ranks.txt", sep = "\t",row.names = TRUE, col.names = NA)


```
Clean up the data in excel - reimport and make a heatmap!

```{r heatmap UIS genes, fig.width = 10}
PvUIS_ranks <- read_excel("outputs/PvUIS_ranks_cleaned.xlsx") %>% 
  mutate(Order_new = 1:24)


colors <- colorRampPalette(c("#FDE725FF","#29AF7FFF", "#2D708EFF", "#440154FF"))(length(1:10))
colors<-c(colors, "black")
```


```{r heatmap, fig.width = 20, fig.height=20}
colour_label<-as_tibble_row(c("PvSPZ.microarray.E.mean.rank" = 1, "spz_bulk_jex" = 2, "Pv1" = 3, "Pv2" = 3, "Pv3" = 3))

colour_label <- t(colour_label) %>% 
  as_tibble(rownames = "rep") %>% 
  mutate(rep = factor(rep, levels =c("PvSPZ.microarray.E.mean.rank", "spz_bulk_jex", "Pv1", "Pv2", "Pv3") ))

as_tibble(colour_label)

tile_plot_gg <- tile_plot %>% 
  select(-value) %>% 
  mutate(decile = as.numeric(decile)) %>% 
  pivot_wider(.,names_from = "rep", values_from = "decile", values_fn = {max}) %>% 
  pivot_longer(-gene, names_to = "rep", values_to = "decile") %>% 
  filter(gene%in%PvUIS_ranks$Gene) %>% 
  left_join(PvUIS_ranks %>% 
            dplyr::rename(gene=Gene)) %>% 
  left_join(Pv_genes %>% 
              dplyr::rename(gene=ID)) %>% 
  mutate(rep = factor(rep, levels =c("PvSPZ.microarray.E.mean.rank", "spz_bulk_jex", "Pv1", "Pv2", "Pv3") )) %>% 
  ggplot(aes(x = as.factor(rep), y = fct_reorder(GeneDescription, -Order_new), fill = factor(decile)))+
  geom_tile(color = "white", size=1)+
  scale_x_discrete(labels = c("", "", "", "", ""))+
  scale_fill_manual(values=setNames(colors, levels(tile_plot$decile)), 
                    na.value = "grey50",
                    labels = c("0-10", "11-20", "21-30", "31-40", "41-50", "51-60", "61-70", "71-80", "81-90", "91-100", "N.D"))+
  labs(fill = "Percentile")+
  tenx_theme()+
  theme(axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        legend.position = "left",
        panel.border = element_blank(),
        axis.ticks =element_blank(), 
        axis.text.x = element_text(angle = 45, hjust=1),
        plot.margin = unit(c(0.2, 1,1,1), "cm"))+
  scale_y_discrete(position = "right")

label <- ggplot(colour_label)+
  geom_tile(mapping = aes(x = rep, y = 1, fill = factor(V1)), stat = "identity", width = 1, colour = "white",size = 1)+
  scale_fill_d3()+
  tenx_theme()+
  scale_fill_discrete(labels = c("Microarray", "Bulk RNA seq", "sc RNA seq"))+
  labs(fill = "", x = "", y = "")+
  theme(plot.margin = unit(c(0,0,0,0), "cm"),
        panel.border = element_blank(),
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank())

fig1g <- label/tile_plot_gg+ plot_layout(heights = c(0.3,10), guides = "collect") &theme(legend.position = "right")


fig1g <- as.ggplot(fig1g)
```

```{r supplementary 3b, fig.width=20, fig.height=10}
corr_comp_pv1 <- all %>% 
  filter(summ_stat == "mean") %>% 
  filter(rep =="Pv1") %>% 
  select(gene, value, type) %>% 
  pivot_wider(names_from = type)

 corr1 <-  ggplot(data = corr_comp_pv1, aes(x = noUTR+0.0001, y = withUTR+0.0001))+
  geom_hex(bins = 10,
           alpha = 0.8)+
  geom_point(alpha = 0.8)+
  geom_smooth(data = filter(corr_comp_pv1, corr_comp_pv1$withUTR>0 & corr_comp_pv1$noUTR>0), 
              method = "lm",
              fullrange = F,
              colour = "red",
              alpha = 0.6)+
  scale_x_log10()+
  scale_y_log10()+
    scale_fill_viridis(guide = guide_colorbar(frame.colour = "black", ticks.colour = "black"), limits = c(0,190))+
    labs(x = "mean expression without UTR", y = "mean expression with UTR", title = "Rep 1")+
    tenx_theme()+
    theme(panel.grid.major = element_line(colour = "grey"))
  
  corr_comp_pv2 <- all %>% 
  filter(summ_stat == "mean") %>% 
  filter(rep =="Pv2") %>% 
  select(gene, value, type) %>% 
  pivot_wider(names_from = type)


corr2 <-   ggplot(data = corr_comp_pv2, aes(x = noUTR+0.0001, y = withUTR+0.0001))+
  geom_hex(bins = 10,
           alpha = 0.8)+
  geom_point(alpha = 0.8)+
  geom_smooth(data = filter(corr_comp_pv2, corr_comp_pv2$withUTR>0 & corr_comp_pv2$noUTR>0), 
              method = "lm",
              fullrange = F,
              colour = "red",
              alpha = 0.6)+
  scale_x_log10()+
  scale_y_log10()+
    scale_fill_viridis(guide = guide_colorbar(frame.colour = "black", ticks.colour = "black"), limits = c(0,190))+
    labs(x = "mean expression without UTR", y = "mean expression with UTR", title = "Rep 2")+
    tenx_theme()+
    theme(panel.grid.major = element_line(colour = "grey"))
  
  corr_comp_pv3 <- all %>% 
  filter(summ_stat == "mean") %>% 
  filter(rep =="Pv1") %>% 
  select(gene, value, type) %>% 
  pivot_wider(names_from = type)


  corr3 <- ggplot(data = corr_comp_pv3, aes(x = noUTR+0.0001, y = withUTR+0.0001))+
  geom_hex(bins = 10,
           alpha = 0.8)+
  geom_point(alpha = 0.8)+
  geom_smooth(data = filter(corr_comp_pv3, corr_comp_pv3$withUTR>0 & corr_comp_pv3$noUTR>0), 
              method = "lm",
              fullrange = F,
              colour = "red",
              alpha = 0.6)+
  scale_x_log10()+
  scale_y_log10()+
    scale_fill_viridis(guide = guide_colorbar(frame.colour = "black", ticks.colour = "black"), limits = c(0,190))+
    labs(x = "mean expression without UTR", y = "mean expression with UTR", title = "Rep 3")+
    tenx_theme()+
    theme(panel.grid.major = element_line(colour = "grey"))
  

s3_layout <- "
ABC
ABC
DDD
"
corr1 + labs(tag = "A") + corr2+corr3 +grid::textGrob('')+labs(tag = "B") + plot_layout(design = s3_layout, guides = 'collect') & theme(plot.tag = element_text(size = 40, face = "bold"))
ggsave("plots/sfig3.tiff", width = 20, height = 10, units = "in",dpi = 320)

```

```{r figure1, fig.width=10, fig.height=10}
layout1 <- "
AAABBBBBBCCC
AAABBBBBB###
DDDDDDDDDDDD
EEGGGGGGGGGG
FFGGGGGGGGGG
FFGGGGGGGGGG
"

map +  plot_spacer()+ mapping_plot + fig_UTR_increase+ cell_count+ grid::textGrob('venn')+ fig1g+plot_layout(design = layout1)+plot_annotation(tag_levels = "A") & theme(plot.tag = element_text(size = 40, face = "bold"))
ggsave("plots/figure1.png", width = 20, height = 24, units = "in")


layout1_withoutmap <- "
AAABBBBBBCCC
DDDDDDDDDDDD
EEGGGGGGGGGG
FFGGGGGGGGGG
"
blank <- ggplot()

blank+plot_spacer()+  mapping_plot + fig_UTR_increase + cell_count+ grid::textGrob('venn')+ fig1g+plot_layout(design = layout1_withoutmap)+plot_annotation(tag_levels = "A") & theme(plot.tag = element_text(size = 40, face = "bold"))
ggsave("plots/figure1_womap.tiff", width = 20, height = 22, units = "in", dpi = 320)
```
End of doc, figure 1 complete.